File: craftground/craftground/csv_logger.py
from datetime import datetime


class CsvLogger:
    def __init__(self, filename, enabled: bool = False):
        self.enabled = enabled
        if enabled:
            self.filename = filename
            self.file = open(filename, "w")
            self.file.write("time,message\n")

    def log(self, message):
        if not self.enabled:
            return
        time_str = datetime.now().strftime("%H:%M:%S.%f")
        self.file.write(f"{time_str},{message}\n")
        self.file.flush()

    def close(self):
        if not self.enabled:
            return
        self.file.close()

File: minecraft/mc121/src/main/java/com/kyhsgeekcode/minecraftenv/CsvLogger.kt
package com.kyhsgeekcode.minecraftenv

import java.io.File

class CsvLogger(
    path: String,
    private val enabled: Boolean = false,
    private val profile: Boolean = false,
) {
    private val file = File(path)
    private val writer = file.bufferedWriter()

    fun log(message: String) {
        if (!enabled) {
            return
        }
        val timestamp = printWithTimeFormatter.format(java.time.LocalDateTime.now())
        writer.write("$timestamp,$message")
        writer.newLine()
        writer.flush()
    }

    fun profileStartPrint(tag: String) {
        if (profile) {
            writer.write("${printWithTimeFormatter.format(java.time.LocalDateTime.now())}, start, $tag\n")
            writer.flush()
        }
    }

    fun profileEndPrint(tag: String) {
        if (profile) {
            writer.write("${printWithTimeFormatter.format(java.time.LocalDateTime.now())}, end, $tag\n")
            writer.flush()
        }
    }

    fun close() {
        writer.close()
    }
}
